server/src/main/java/com/sauron/service/TransactionServiceImpl.java

```diff
@@ -1,40 +1,54 @@
 package com.sauron.service;
 
 import com.sauron.model.Transaction;
+import com.sauron.model.views.BankView;
+import com.sauron.model.views.UserView;
 import org.springframework.boot.web.client.RestTemplateBuilder;
 import org.springframework.core.ParameterizedTypeReference;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.RequestEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponents;
 
-import java.net.URI;
 import java.util.Collection;
-import java.util.List;
 import java.util.stream.Collectors;
 
+import static org.springframework.web.util.UriComponentsBuilder.fromHttpUrl;
+
 @Service
 public class TransactionServiceImpl implements TransactionService {
 
-    private static final String SHIRE_BANK_TRANSACTIONS_URL = "http://localhost:8080/fake/transactions/shire-bank";
-    private static final String MORDOR_BANK_TRANSACTIONS_URL = "http://localhost:8080/fake/transactions/mordor-bank";
-    private static final List<String> BANK_URLS = List.of(SHIRE_BANK_TRANSACTIONS_URL, MORDOR_BANK_TRANSACTIONS_URL);
+    private static final String USER_ID_PARAM = "userId";
 
+    private final UserService userService;
     private final RestTemplate restTemplate;
 
-    public TransactionServiceImpl(RestTemplateBuilder restTemplateBuilder) {
+    public TransactionServiceImpl(RestTemplateBuilder restTemplateBuilder, UserService userService) {
         this.restTemplate = restTemplateBuilder.build();
+        this.userService = userService;
     }
 
     @Override
-    public Collection<Transaction> getAllTransactions() {
-
-        return BANK_URLS.stream().flatMap(url -> getBankTransactions(url).stream()).collect(Collectors.toList());
+    public Collection<Transaction> getAllTransactions(final Long userId) {
+        UserView user = userService.get(userId);
```

Comment:
As for me UserView and BankView are classes for UI purpose only. You shouldn't use them in business logic. Just use repositories directly, you will get User and Bank instances respectively.